### PR TITLE
feat: added prometheus support and histogram metrics

### PR DESCRIPTION
**Pull request by blockchainluffy**
Added prometheus support with server metrics and histogram metrics for:

- total request duration
- duration of encryption
- requested gas limit

### CODE REVIEW

File: src/main.go [hunk 1]
```diff
@@ -118,6 +123,30 @@ func Cmd() *cobra.Command {
...
 
+	cmd.PersistentFlags().BoolVarP(
+		&Config.MetricsConfig.Enabled,
+		"metrics-enabled",
+		"",
+		false,
+		"to enable promnetheus metrics",
+	)
+
+	cmd.PersistentFlags().StringVarP(
```
Comment (ylembachar) [inline src/main.go:134]: would be good to set the default values here for host and port to the default prometheus ones? local host and 9090?

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+services:
```
Comment (ylembachar) [inline docker-compose.yml:1]: could you please add this service to the main docker compose file as we already have one with all the services under the docker package?

File: src/rpc/service_eth.go [hunk 3]
```diff
@@ -241,13 +246,17 @@ var DefaultProcessTransaction = func(tx *txtypes.Transaction, ctx context.Contex
...
+	timeBefore := time.Now()
+
 	identityPrefix, err := shcrypto.RandomSigma(cryptorand.Reader)
 	if err != nil {
 		return nil, &EncodingError{StatusCode: -32602, Err: err}
 	}
 	identity := ComputeIdentity(identityPrefix[:], newSigner.From)
 	encryptedTx := shcrypto.Encrypt(b, eonKey, identity, sigma)
 
+	metrics.MetricsEncryptionDuration.WithLabelValues(tx.Hash().String()).Observe(float64(time.Since(timeBefore).Seconds()))
```
Comment (ylembachar) [inline src/rpc/service_eth.go:258]: conversion to float64 could be removed here

File: src/rpc/service_eth.go [hunk 4]
```diff
@@ -207,6 +210,8 @@ func (service *EthService) SendRawTransaction(ctx context.Context, s string) (*c
 	}
 	utils.Logger.Info().Hex("Incoming tx hash", txHash.Bytes()).Hex("Encrypted tx hash", submitTx.Hash().Bytes()).Msg("Transaction sent")
 
+	metrics.MetricsRequestedGasLimit.WithLabelValues(txHash.String()).Observe(float64(tx.Gas()))
+	metrics.MetricsTotalRequestDuration.WithLabelValues(txHash.String()).Observe(float64(time.Since(timeBefore)))
```
Comment (ylembachar) [inline src/rpc/service_eth.go:214]: time here is in nanoseconds, would be great to update it to ms as more readable and clear

File: src/rpc/service_eth.go [hunk 5]
```diff
@@ -241,13 +246,17 @@ var DefaultProcessTransaction = func(tx *txtypes.Transaction, ctx context.Contex
...
+	timeBefore := time.Now()
+
 	identityPrefix, err := shcrypto.RandomSigma(cryptorand.Reader)
 	if err != nil {
 		return nil, &EncodingError{StatusCode: -32602, Err: err}
 	}
 	identity := ComputeIdentity(identityPrefix[:], newSigner.From)
 	encryptedTx := shcrypto.Encrypt(b, eonKey, identity, sigma)
 
+	metrics.MetricsEncryptionDuration.WithLabelValues(tx.Hash().String()).Observe(time.Since(timeBefore).Seconds())
```
Comment (ylembachar) [inline src/rpc/service_eth.go:258]: maybe also here have it in ms as it's more uniform and readable

**Review by ylembachar — APPROVED** [review]

File: docker/docker-compose.yaml [hunk 6]
```diff
@@ -32,5 +34,21 @@ services:
...
+
+  vm:
+    image: victoriametrics/victoria-metrics
+    restart: always
+    command: -promscrape.config /config/prometheus.yml
+    ports:
+      - "8428:8428"
+    volumes:
+      - ../data/victoriametrics:/victoria-metrics-data
+      - ../config:/config
```
Comment (ulope) [inline docker/docker-compose.yaml:50]: Sorry I think we had a miscommunication here.
We should scrape the metrics from our already existing VictoriaMetrics installation. 
Doesn't really make too much sense to add another metrics collection point (at least not at the scale we're currently at).

It also doesn't necessarily hurt to leave it in but I think at this point it just adds complexity for not much gain.

File: docker/docker-compose.yaml [hunk 7]
```diff
@@ -14,6 +14,8 @@ services:
       --keyper-set-manager-address ${KEYPER_SET_MANAGER_CONTRACT_ADDRESS}
       --rpc-url ${UPSTREAM_RPC}
       --delay-in-seconds ${DELAY_IN_SECONDS}
+    networks:
+      - app-network
```
Comment (ulope) [inline docker/docker-compose.yaml:18]: This isn't necessary. Compose creates a default network that all containers of a project are connected to by default.

**Review by ulope — APPROVED** [review]
LGTM

Apart from the things mentioned below.

**Pull request merged**